astronet/astro_cnn_model/input_ds.py

```diff
@@ -93,6 +95,116 @@ def parse_example(serialized_example):
     ds = tf.data.Dataset.from_tensor_slices(filenames)
     ds = ds.flat_map(tf.data.TFRecordDataset)
     ds = ds.map(parse_example)
+
+###### rough code for checking whether the candidate has been assigned disps E,N,B, or S, and only using these candidates in the data augmentation. There is probably a more concise way to do this ######
+
+
+#     values = np.array([x[1].numpy() for x in ds])
+#    truth_array = []
+#    for array in values:
+#        z = array == [1., 1., 1., 1., 1.]
+#        if z[0] or z[1] or z[3] or z[4]:
+#            truth_array.append(True)
+#        else:
+#            truth_array.append(False)       
+#    truth_data = tf.data.Dataset.from_tensor_slices(truth_array)
+#    ds_new = tf.data.Dataset.zip((ds,truth_data))
+#    new_len = [x for x in ds_new]
+#    print("number trues =",np.array(truth_array).sum())
+#    ds_false = ds_new.filter(lambda image,y:y==False)
+#    false_list = [x for x in ds_false]
+#    randoms = []
+#    for i in range(len(false_list)):
+#        randoms.append(random.randint(0,1))
+#    rand = tf.data.Dataset.from_tensor_slices(randoms)
+#    false_rand = tf.data.Dataset.zip((ds_false,rand))
+#    false_rand = false_rand.filter(lambda x,z: z==1)
+#    ds_new = ds_new.filter(lambda image,y: y ==True)
+    
+    
+    
+    globals  = ['global_mask', 'global_std', 'global_transit_mask', 'global_view', 'global_view_0.3', 'global_view_5.0', 'global_view_double_period', 'global_view_double_period_0.3', 'global_view_double_period_5.0']
+    locals   =['local_mask','local_std','local_std_even','local_std_odd','local_view','local_view_0.3','local_view_5.0','local_view_even','local_view_half_period_std','local_view_odd']
+    secondaries = ['secondary_mask','secondary_std','secondary_view','secondary_view_0.3','secondary_view_5.0']
+    all  = ['global_mask', 'global_std', 'global_transit_mask', 'global_view', 'global_view_0.3', 'global_view_5.0', 'global_view_double_period', 'global_view_double_period_0.3', 'global_view_double_period_5.0','local_mask','local_std','local_std_even','local_std_odd','local_view','local_view_0.3','local_view_5.0','local_view_even','local_view_half_period_std','local_view_odd','secondary_mask','secondary_std','secondary_view','secondary_view_0.3','secondary_view_5.0']
+    
+
+    def add_noise_global(image,y):
+        for a in globals: 
+            z = image[0][a]
+            noise = tf.random.normal(shape=z.get_shape(), mean=0.0, stddev=0.1, dtype=tf.float32)
+            image[0][a] = tf.math.add(z,noise)
+        return image[0],image[1],image[2]
+    
+    def add_noise_local(image,y):
+        for a in locals: 
+            z = image[0][a]
+            noise = tf.random.normal(shape=z.get_shape(), mean=0.0, stddev=0.1, dtype=tf.float32)
```

Comment:
Hmm these noise arrays are independently generated, when in reality noise between views is correlated.

Maybe for now we just say this doesn't matter too much